pkg/detectors/cloudflarecakey/cloudflarecakey.go

```diff
@@ -54,13 +54,19 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			req.Header.Add("Content-Type", "application/json")
 			req.Header.Add("user-agent", "curl/7.68.0") // pretend to be from curl so we do not wait 100+ seconds -> nice try did not work
 
-			req.Header.Add("X-Auth-User-Service-Key", resMatch)
-			res, err := client.Do(req)
+			req.Header.Add("X-Auth-User-Service-Key", caKey)
+			resp, err := client.Do(req)
 			if err == nil {
-				defer res.Body.Close()
-				if res.StatusCode >= 200 && res.StatusCode < 300 {
+				defer func() {
```

Comment:
The whole verify block should be moved into its own function. It's both cleaner and Defer in for loops isn't ideal.

Reply:
Done ✅